common/config/src/main/java/io/helidon/common/config/Config.java

```diff
@@ -0,0 +1,346 @@
+     *
+     * @param key sub-key of requested sub-node
+     * @return config node for specified sub-key, never returns {@code null}.
+     * @see #get(Config.BaseKey)
+     * @throws io.helidon.common.config.ConfigException if not defined
+     */
+    default Config get(String key) throws ConfigException {
+        throw new ConfigException("unsupported; do you have a valid config module in the classpath?");
+    }
+
+    /**
+     * Returns the single sub-node for the specified sub-key.
+     *
+     * @param key sub-key of requested sub-node
+     * @return config node for specified sub-key, never returns {@code null}.
+     * @see #get(String)
+     */
+    default Config get(BaseKey key) {
+        return get(key.name());
+    }
+
+    /**
+     * Returns a copy of the {@code Config} node with no parent.
+     * <p>
+     * The returned node acts as a root node for the subtree below it. Its key
+     * is the empty string; {@code ""}. The original config node is unchanged,
+     * and the original and the copy point to the same children.
+     * <p>
+     * Consider the following configuration:
+     * <pre>
+     * app:
+     *      name: Example 1
+     *      page-size: 20
+     * logging:
+     *      app.level = INFO
+     *      level = WARNING
+     * </pre>
+     * The {@code Config} instances {@code name1} and {@code name2} represents same data and
+     * in fact refer to the same object:
+     * <pre>
+     * Config name1 = config
+     *                  .get("app")
+     *                  .get("name");
+     * Config name2 = config
+     *                  .get("app")
+     *                  .detach()               //DETACHED node
+     *                  .get("name");
+     *
+     * assert name1.asString() == "Example 1";
+     * assert name2.asString() == "Example 1";  //DETACHED node
+     * </pre>
+     * The only difference is the key each node returns:
+     * <pre>
+     * assert name1.key() == "app.name";
+     * assert name2.key() == "name";            //DETACHED node
+     * </pre>
+     * <p>
+     * See asMap() for example of config detaching.
+     *
+     * @return returns detached Config instance of same config node
+     * @throws io.helidon.common.config.ConfigException if not defined
+     */
+    default Config detach() throws ConfigException {
+        throw new ConfigException("unsupported; do you have a valid config module in the classpath?");
+    }
+
+    /**
+     * Returns {@code true} if the node exists, whether an object, a list, a
+     * value node, etc.
+     *
+     * @return {@code true} if the node exists
+     */
+    boolean exists();
+
+    /**
+     * Returns {@code true} if this node exists and is a leaf node (has no
+     * children).
+     * <p>
+     * A leaf node has no nested configuration subtree and has a single value.
+     *
+     * @return {@code true} if the node is existing leaf node, {@code false}
+     *         otherwise.
+     */
+    boolean isLeaf();
```

Comment:
Not really needed. This is basically `!isObject() && !isKey() && hasValue()`.
Also method `hasValue()` is missing

Reply:
There is no isKey() on either common/config/Config or config/config.  I will add hasValue() decl on common/config.

Reply:
sorry, meant `isList()` not `isKey()`